File: angular-front-end/src/app/model/cart-item.ts
import { Product } from './product';

export class CartItem {
  id: string | null = null;
  quantity: number = 0;
  addedAt: Date = new Date();
  price: number = 0;
  product: Product = new Product();
  userId: number = 0;
  constructor() {
  }
}

File: catalogue-service/src/main/java/org/nc/entity/CartItem.java
package org.nc.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Data @AllArgsConstructor @NoArgsConstructor
public class CartItem {
    @Id
    private String id;
    private int quantity;
    private Date addedAt;
    private double price;
    private Long userId;

    @DBRef
    private Product product;

}
